src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,104 @@
+package duke.command;
+
+import duke.Duke;
+// import duke.command.method.*;
+import duke.command.method.ByeCommandMethod;
+import duke.command.method.CommandMethod;
+import duke.command.method.DeadlineCommandMethod;
+import duke.command.method.DeleteCommandMethod;
+import duke.command.method.DoneCommandMethod;
+import duke.command.method.EventCommandMethod;
+import duke.command.method.FindCommandMethod;
+import duke.command.method.ListCommandMethod;
+import duke.command.method.TodoCommandMethod;
+import duke.exception.DukeException;
+import duke.exception.DukeNoCommandException;
+import duke.exception.DukeUnrecognisedCommandException;
+
+public class Command {
+    private String name;
+    private String[] arguments;
+    private CommandMethod method;
+
+    public static Command createCommand(String input) throws DukeException {
+        String name;
+        String[] arguments;
+        if (input.contains(" ")) {
+            String[] inputs = input.split(" ", 2);
+            name = inputs[0];
+            if (inputs[1].contains(" ")) {
+                arguments = inputs[1].split(" ");
+            } else {
+                arguments = new String[] {inputs[1]};
+            }
+        } else {
+            name = input;
+            arguments = new String[0];
+        }
+        switch (name) {
+        case "": {
+            throw new DukeNoCommandException();
+        }
+        case TodoCommandMethod.NAME: {
+            return new Command(name, arguments, new TodoCommandMethod());
+        }
+        case EventCommandMethod.NAME: {
+            return new Command(name, arguments, new EventCommandMethod());
+        }
+        case DeadlineCommandMethod.NAME: {
+            return new Command(name, arguments,
+                    new DeadlineCommandMethod());
+        }
+        case ListCommandMethod.NAME: {
+            return new Command(name, arguments, new ListCommandMethod());
+        }
+        case DoneCommandMethod.NAME: {
+            return new Command(name, arguments, new DoneCommandMethod());
+        }
+        case DeleteCommandMethod.NAME: {
+            return new Command(name, arguments, new DeleteCommandMethod());
+        }
+        case FindCommandMethod.NAME: {
+            return new Command(name, arguments, new FindCommandMethod());
+        }
+        case ByeCommandMethod.NAME: {
+            return new Command(name, arguments, new ByeCommandMethod());
+        }
```

Comment:
I like how each of the individual commands are being refactored into their own class. It lessens the workload of the Command class and make the code more readable. Good job!